otter/convergence/selfheal.py

```diff
@@ -78,103 +72,36 @@ def _cancel_scheduled_calls(self):
         :return: Number of remaining active scheduled calls before cancelling
         """
         active = 0
-        for call in self.calls:
+        for call in self._calls:
             if call.active():
                 active += 1
                 call.cancel()
-        self.calls = []
+        self._calls = []
         return active
 
-    def health_check(self):
-        """
-        Return about whether this object has lock
-        """
-        d = self.lock.is_acquired()
-        return d.addCallback(lambda b: (True, {"has_lock": b}))
-
     @inlineCallbacks
-    def _setup_convergences(self, config_func, time_range):
+    def _setup_convergences(self):
         """
         Get groups to converge and setup scheduled calls to trigger convergence
-        on each of them within time_range. For parameters, see
-        :func:`__init__` docs.
+        on each of them within time_range.
         """
-        groups = yield perform(self.disp,
-                               get_groups_to_converge(config_func))
+        groups = yield perform(self.dispatcher,
+                               get_groups_to_converge(self.config_func))
         active = self._cancel_scheduled_calls()
         if active:
             # This should never happen
-            self.log.err(RuntimeError("self-heal-calls-err"),
-                         "self-heal-calls-err", active=active)
+            self.log.err(RuntimeError("selfheal-calls-err"),
```

Comment:
I went through lot of iterations of this code as you can see in commits and decided to use this ID probably because I was generating this ID instead of hard-coding it at one time. In any case, this seemed more appealing due to its one less character :). However, thanks to you I just realized that this to updated in `spec.py` where it is expanded.